test/smartwallet/SmartWallet.test.ts

```diff
@@ -550,26 +548,38 @@ options.forEach(element => {
           const func = recipient.contract.methods.emitMessage('hello').encodeABI()
 
           const initialNonce = await sw.nonce()
-          await sw.directExecute(recipient.address, func, { from: otherAccount })
+          await sw.directExecute(recipient.address, 0, func, { from: otherAccount })
 
           // @ts-ignore
           const logs = await recipient.getPastEvents('TestForwarderMessage')
+
           assert.equal(logs.length, 1, 'TestRecipient should emit')
           assert.equal(logs[0].args.origin, otherAccount, 'test "from" account is the tx.origin')
           assert.equal(logs[0].args.msgSender, sw.address, 'msg.sender must be the smart wallet address')
 
           assert.equal((await sw.nonce()).toString(), initialNonce.toString(), 'direct execute should NOT increment nonce')
         })
 
+        it('should fail if executed function is non-payable', async () => {
+          const func = recipient.contract.methods.emitMessage('hello').encodeABI()
```

Comment:
Just checked and in my understanding, this function shouldn't fail because it's payable. What Am I missing?